numba/parfors/parfor_lowering.py

```diff
@@ -52,6 +52,15 @@ def lower_inst(self, inst):
         else:
             super().lower_inst(inst)
 
+    @property
+    def _disable_sroa_like_opt(self):
+        """
+        Force disable this because Parfor use-defs is incompatible---it only
+        considers use-defs in blocks that must be executing.
+        See https://github.com/numba/numba/commit/017e2ff9db87fc34149b49dd5367ecbf0bb45268
+        """
+        return True
```

Comment:
After thought on this. It's possible that this may increase compilation times, however it prevents miscompilation. It's better to work from a position of correctness towards performance...!